src/cpp/fastdds/publisher/DataWriterImpl.hpp

```diff
@@ -388,6 +391,10 @@ class DataWriterImpl : protected rtps::IReaderDataFilter
     void filter_is_being_removed(
             const char* filter_class_name);
 
+private:
+
+    bool is_custom_payloadpool = false;
```

Comment:
Since `payload_pool_` is protected, I would also make this one protected. And I would have it next to `payload_pool_`.

Also: 
```suggestion
    bool is_custom_payload_pool_ = false;
```